src/app/components/NetworkRibbon/component/DetectionScreen.tsx

```diff
@@ -24,9 +26,22 @@ interface Props {
 
 export function DetectionScreen(props: Props) {
   var logo: any = null;
-  const { ethereum } = window;
   const { t } = useTranslation();
-  const chainId = parseInt(ethereum?.chainId as string);
+
+  const { wallet, expectedChainId } = useContext(WalletContext);
+
+  const chainId = useMemo(() => {
+    if (isWeb3Wallet(wallet.providerType!)) {
+      const provider = InjectedWalletProvider.getProvider(
+        expectedChainId || currentChainId,
+      );
+      if (providers) {
+        return parseInt(provider.chainId);
+      }
+    }
+    return parseInt(wallet?.chainId.toString());
```

Comment:
Why do you need this conversion? `wallet.chainId` is a `number` already so I think we're ok without `parseInt` and `toString()`.